fellow-submissions/stephanie-smith.md

```diff
@@ -0,0 +1,108 @@
+# Problem Set 2.4 - Scope
+
+**1. How do variables declared with `let`, `const`, and `var` differ? Be sure to touch on _scope_, _reassignment_, and _hoisting_ for each.**
+- Variables declared with `var` can be reassigned as often as you want, and you can access them regardless of scope. However, if you are going to use `var` with function expressions, you cannot hoist them, so you'll have to declare it after the function expression is initialized.
+  Variables declared with `let` can also be reassigned, but you cannot access them outside their scopes, so you can use the same variable in different scopes and get different results. Regarding hoisting, they also cannot be hoisted so you'll have to declare them after they're initialized. Variables declared with `const` cannot be reassigned, so whatever is initialized in the variable remains that way. They also cannot be hoisted, and just like `let`, they cannot be accessed outside their scope.
+
+**2. What will the following code log? Why?**
+    ```javascript
+    let a = 'outer';
+
+    function testScope() {
+      let a = 'inner';
+      console.log(a);
+    }
+
+    console.log(a);
+    testScope();
+    console.log(a);
+    ```
+  - It will first log 'outer', then it will log 'inner', then it will log 'outer' again. The global scope will be called first because there is no direct reference to the scope that 'inner' is in. However, when the function is called, `a`'s value changes into what it is in the scope, and logs that out instead. Then it refers back to the global scope.
+
+**3. What will the following code log? Why?**
+    ```javascript
+    function hello() {
+      let a = 'hello';
+    }
+
+    hello();
+    console.log(a);
+    ```
+  - It will log 'outer', then 'a is not defined' , because the function `hello()` was called first, allowing the variable `a` to log its value. However, when you console.log `a` without calling the function, it does not work because there is no `a` in the global scope. Since the `a` in the function can't be accessed from the global scope, it simply remains undefined because `a` technically does not exist.
```

Comment:
Try this one again. `outer` was not a part of this question. And the function `hello` does not log anything. Your understanding of scope is correct here but test this in the console to be sure of what it logs. Then tell me why.